odiglet/pkg/ebpf/manager.go

```diff
@@ -0,0 +1,371 @@
+	instrumentationConfigKey := client.ObjectKey{
+		Namespace: w.Namespace,
+		Name:      workloadUtils.CalculateWorkloadRuntimeObjectName(w.Name, w.Kind),
+	}
+	if err := m.Client.Get(ctx, instrumentationConfigKey, &instrumentationConfig); err != nil {
+		// this can be valid when the instrumentation config is deleted and current pods will go down soon
+		m.Logger.Error(err, "failed to get initial instrumentation config for instrumented pod", "pod", podKey.Name, "namespace", podKey.Namespace)
+		return nil
+	}
+	for _, config := range instrumentationConfig.Spec.SdkConfigs {
+		if config.Language == lang {
+			return &config
+		}
+	}
+	return nil
+}
+
+func (m *Manager) applyInstrumentationConfigurationForSDK(ctx context.Context, workloadKey types.NamespacedName, sdkConfig *odigosv1.SdkConfig) error {
+	var err error
+	for _, instDetails := range m.DetailsByPid {
+		if instDetails.WorkloadName != workloadKey.Name || instDetails.WorkloadNamespace != workloadKey.Namespace {
+			continue
+		}
+
+		if instDetails.Lang != sdkConfig.Language {
+			continue
+		}
+
+		m.Logger.Info("applying configuration to instrumentation", "workload", workloadKey, "pid", instDetails.Inst, "pod", instDetails.Pod)
+		err = errors.Join(err, instDetails.Inst.ApplyConfig(ctx, sdkConfig))
+	}
+	return err
+}
+
+func (m *Manager) updateInstrumentationInstanceStatus(ctx context.Context, pod *corev1.Pod, containerName string, w *workloadUtils.PodWorkload, pid int, health InstrumentationHealth, reason InstrumentationStatusReason, msg string) error {
+	instrumentedAppName := workloadUtils.CalculateWorkloadRuntimeObjectName(w.Name, w.Kind)
+	healthy := bool(health)
+	return instance.UpdateInstrumentationInstanceStatus(ctx, pod, containerName, m.Client, instrumentedAppName, pid, m.Client.Scheme(),
+		instance.WithHealthy(&healthy, string(reason), &msg),
+	)
+}
+
+func (m *Manager) podFromProcEvent(event detector.ProcessEvent) (*corev1.Pod, error) {
+	eventEnvs := event.ExecDetails.Environments
+
+	podName, ok := eventEnvs[consts.OdigosEnvVarPodName]
+	if !ok {
+		return nil, fmt.Errorf("missing %s in environment variables", consts.OdigosEnvVarPodName)
+	}
+
+	podNamespace, ok := eventEnvs[consts.OdigosEnvVarNamespace]
+	if !ok {
+		return nil, fmt.Errorf("missing %s in environment variables", consts.OdigosEnvVarNamespace)
+	}
+
+	pod := corev1.Pod{}
+	// TODO: pass context from outer function
+	err := m.Client.Get(context.Background(), client.ObjectKey{Namespace: podNamespace, Name: podName}, &pod)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching pod object: %w", err)
+	}
+
+	return &pod, nil
+}
+
+func containerNameFromProcEvent(event detector.ProcessEvent) (string, bool) {
+	containerName, ok := event.ExecDetails.Environments[consts.OdigosEnvVarContainerName]
+	return containerName, ok
+}
+
+func (m *Manager) languageSdk(pod *corev1.Pod, containerName string) (common.ProgrammingLanguage, common.OtelSdk, error) {
```

Comment:
suggestion for alternative name
```suggestion
func (m *Manager) getContainerDeviceInfo(pod *corev1.Pod, containerName string) (common.ProgrammingLanguage, common.OtelSdk, error) {
```

Reply:
Moved to `k8sutils`, and renamed `LanguageSdkFromPodContainer`